react/src/components/PortalAccessKey.js

```diff
@@ -0,0 +1,116 @@
+import { useState } from 'react';
+import Auth from '../utils/Auth';
+import Char from '../utils/Char';
+import Time from '../utils/Time';
+import useHeader from '../hooks/Header';
+
+const PortalAccessKey = (props) => {
+    const header = useHeader();
+    const accessKey = props.accessKey;
+    const color = props.error ? "darkred" : "inhert";
+    const tdstyle = { fontSize: "11pt", color: color, verticalAlign: "top", paddingBottom: "2pt", paddingRight: "10pt", whiteSpace: "nowrap" };
+    const tdlabel = { ...tdstyle, width: "1%" };
+    const [ showDetails, setShowDetails ] = useState(false);
+    const [ showJson, setShowJson ] = useState(false);
+    if (!accessKey) return <></>
+    return <>
+        <table width="100%"><tbody>
+            <tr>
+                <td style={tdlabel}>Portal:</td>
+                <td style={tdstyle}>
+                    <a href={`${accessKey.server}`} target="_blank">{accessKey.server}</a>
+                </td>
+            </tr>
+            <tr>
+                <td style={tdlabel}>Access Key:</td>
+                <td style={tdstyle}>
+                    {accessKey.key}
+                    { Auth.IsLoggedIn(header) && <>
+                        &nbsp;&nbsp;{Char.RightArrow}&nbsp;&nbsp;
+                        <a href={`${accessKey.server}/access-keys/${accessKey.key}/`} style={{color:color}} target="_blank">View</a>
+                    </> }
+                    <small style={{float:"right", marginRight:"-10pt"}}><b>
+                        { showJson ? <>
+                            <span onClick={() => setShowJson(false)} style={{cursor:"pointer"}}>
+                                JSON&nbsp;{Char.DownArrow}
+                            </span>
+                        </>:<>
+                            <span onClick={() => setShowJson(true)} style={{cursor:"pointer"}}>
+                                JSON&nbsp;{Char.UpArrow}
+                            </span>
+                        </> }
+                    </b></small>
+                </td>
+            </tr>
+            <tr>
+                <td style={tdlabel}>Access Key Secret:</td>
+                <td style={tdstyle}>
+                    {accessKey.secret}
+                </td>
+            </tr>
```

Comment:
Let's not display this value... the key ID should be enough, doing this adds great risk of exposure of this key I'd say, and is especially dangerous because it is not tied to any one user (Foursight generally).

Reply:
Done (as discussed also on Slack).